### ISSUE TITLE
details component - aria-expanded not announced by Voiceover on Safari

### ISSUE THREAD

**Issue by kata-lin** [Legal-and-General/canopy#726]
The aria-expanded: true value for the details component is not being announced by Voiceover in Safari (Version 14.1.2). This is correctly announced in Chrome.

Working example of the aria-expanded value on a button can be found here, https://pauljadam.com/demos/button.html. 

The only difference that might be causing the different behaviour is the nesting.

**Comment by elenagarrone** [comment]
Might be related to the fact that the aria attributes are set incorrectly:
![Screenshot 2022-08-12 at 16 09 01](https://user-images.githubusercontent.com/8397116/184384977-f7e82c3b-931d-4ab9-9ce7-6309acb064ef.png)

**Comment by elenagarrone** [comment]
The `id` and `aria-labelledby` are now set correctly. 
I'm closing this as it seem to be more of a bug with Safari rather than the structure of the component. 

The `aria-expanded` is read correctly when tabbing to the element, it's just not read when toggling open/closed.
